File: 268_Missing_Number.c
#include<stdio.h>
#include<stdlib.h>

int missingNumber(int* nums, int numsSize) {
    int mis_num = 0;
    for (int i = 0; i < numsSize; i++) {
        mis_num ^= (i + 1) ^ nums[i];
    }

    return mis_num;
}

int main() {
    int nums[] = {3, 0, 1, 2};
    int numsSize = sizeof(nums) / sizeof(nums[0]);

    int missing = missingNumber(nums, numsSize);
    printf("Missing number: %d\n", missing);

    return 0;
}

File: 268_Missing_Number.py
class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        mis_num = 0
        limit = len(nums)

        for i in range(limit+1):
            mis_num ^= i

        for i in nums:
            mis_num ^= i

        return mis_num


a = Solution()
print(a.missingNumber([3,0,1])) # 2
print(a.missingNumber([0,1])) # 2
print(a.missingNumber([9,6,4,2,3,5,7,0,1])) # 8
        